packages/schema-blocks/src/functions/validators/innerBlocksValid.ts

```diff
@@ -30,19 +31,19 @@ function findMissingBlocks( requiredBlocks: RequiredBlock[], existingRequiredBlo
 /**
  * Finds all blocks that occur more than once in the inner blocks.
  *
- * @param requiredBlocks         Requirements of the blocks that should occur only once in the inner blocks.
  * @param existingRequiredBlocks The actual array of all inner blocks.
+ * @param requiredBlocks         Requirements of the blocks that should occur only once in the inner blocks.
  *
  * @returns {InvalidBlock[]} The names of blocks that occur more than once in the inner blocks with reason 'TooMany'.
  */
-function findRedundantBlocks( requiredBlocks: RequiredBlock[], existingRequiredBlocks: BlockInstance[] ): InvalidBlock[] {
+function findRedundantBlocks( existingRequiredBlocks: BlockInstance[], requiredBlocks: RequiredBlock[] ): InvalidBlock[] {
 	const onlyOneAllowed: string[] = [];
 	const invalidBlocks: InvalidBlock[] = [];
 
 	requiredBlocks
 		.filter( block => block.option === RequiredBlockOption.One )
 		.forEach( block => onlyOneAllowed.push( block.name ) );
```

Comment:
The `forEach` is not necessary here. The `filter` function already contains all the `onlyOneAllowed` blocks.
```suggestion
	onlyOneAllowed = requiredBlocks
		.filter( block => block.option === RequiredBlockOption.One );
```